controllers/components/codeflare/codeflare_controller.go

```diff
@@ -19,40 +19,69 @@ package codeflare
 import (
 	"context"
 
-	"k8s.io/apimachinery/pkg/runtime"
+	promv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/render/kustomize"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/updatestatus"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/predicates/resources"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/reconciler"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
 )
 
 // CodeFlareReconciler reconciles a CodeFlare object.
-type CodeFlareReconciler struct {
-	client.Client
-	Scheme *runtime.Scheme
-}
 
-// Reconcile is part of the main kubernetes reconciliation loop which aims to
-// move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the CodeFlare object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
-func (r *CodeFlareReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
-
-	// TODO(user): your logic here
-
-	return ctrl.Result{}, nil
-}
+func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	_, err := reconciler.ComponentReconcilerFor(
+		mgr,
+		componentsv1.CodeFlareInstanceName,
+		&componentsv1.CodeFlare{},
+	).
+		// customized Owns() for Component with new predicates
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.Secret{}).
+		Owns(&rbacv1.ClusterRoleBinding{}).
+		Owns(&rbacv1.ClusterRole{}).
+		Owns(&rbacv1.Role{}).
+		Owns(&rbacv1.RoleBinding{}).
+		Owns(&corev1.ServiceAccount{}).
+		Owns(&corev1.Service{}).
+		Owns(&networkingv1.NetworkPolicy{}).
+		Owns(&promv1.PodMonitor{}).
+		Owns(&promv1.PrometheusRule{}).
+		Owns(&admissionregistrationv1.MutatingWebhookConfiguration{}).
+		Owns(&admissionregistrationv1.ValidatingWebhookConfiguration{}).
+		Owns(&appsv1.Deployment{}, reconciler.WithPredicates(resources.NewDeploymentPredicate())).
+		Watches(&extv1.CustomResourceDefinition{}).
```

Comment:
Looking at the manifests, it seems they include the following GVKs:
```
apiextensions.k8s.io/v1: CustomResourceDefinition
rbac.authorization.k8s.io/v1: Role
rbac.authorization.k8s.io/v1: ClusterRole
rbac.authorization.k8s.io/v1: RoleBinding
rbac.authorization.k8s.io/v1: ClusterRoleBinding
v1: ConfigMap
v1: Secret
v1: Service
v1: ServiceAccount
apps/v1: Deployment
admissionregistration.k8s.io/v1: MutatingWebhookConfiguration
admissionregistration.k8s.io/v1: ValidatingWebhookConfiguration
```

I may be wrong, but I don't see `NetworkPolicy`, `PodMonitor` and `PrometheusRule`

Reply:
@lburgazzoli I thought through ```
Owns(&networkingv1.NetworkPolicy{}).
		Owns(&promv1.PodMonitor{}).
		Owns(&promv1.PrometheusRule{})
```
that would be added to the manifests. Am I missing something or should be updating something else as well?

Reply:
I meant that I don't see `NetworkPolicy`, `PodMonitor` and `PrometheusRule` in the YAML files provided by codeflare in `opt/manifests/codeflare` so I don't think we should include them as part of the resource types owned by the controller so I think we can remove those lines. However

Reply:
Removed.